sdk/web-pubsub/web-pubsub/review/web-pubsub.api.md

```diff
@@ -90,24 +130,27 @@ export interface HubSendToUserOptions extends OperationOptions {
 }
 
 // @public
-export class WebPubSubGroup {
-    // @internal
-    constructor(hubName: string, groupName: string);
+export type JSONTypes = string | number | boolean | object;
+
+// @public (undocumented)
+export type Permission = "joinLeaveGroup" | "sendToGroup";
+
+// @public (undocumented)
+export interface WebPubSubGroup {
     addConnection(connectionId: string, options?: GroupAddConnectionOptions): Promise<RestResponse>;
     addUser(username: string, options?: GroupAddUserOptions): Promise<RestResponse>;
     readonly apiVersion: string;
-    endpoint: string;
+    readonly endpoint: string;
     readonly groupName: string;
     hasUser(username: string, options?: GroupHasUserOptions): Promise<boolean>;
     readonly hubName: string;
     removeConnection(connectionId: string, options?: GroupRemoveConnectionOptions): Promise<RestResponse>;
     removeUser(username: string, options?: GroupRemoveUserOptions): Promise<RestResponse>;
-    // Warning: (ae-forgotten-export) The symbol "GroupSendTextToAllOptions" needs to be exported by the entry point index.d.ts
     sendToAll(message: string, options: GroupSendTextToAllOptions): Promise<RestResponse>;
```

Comment:
Thinking about this... I kinda don't like writing 

```ts
client.sendToAll("some text", { contentType: "text/plain" });
```

It feels like the type of the message should be part of the message parameter, like

```
sendToAll(message: JSONTypes | { plainText: string } | HttpRequestBody): Promise<RestResponse>
```

This also gets rid of needing `GroupSendTextToAllOptions`

Reply:
I don't like this, because this is not part of the message. The first parameter should be more or less exactly what gets sent to clients on the websocket.

Reply:
I'm thinking of specifically of how `data-tables` handles denoting ambiguous primitive types for things like guids and int64:

https://github.com/Azure/azure-sdk-for-js/blob/7aad1d340bf6b7b805804c961f789d94d286c069/sdk/tables/data-tables/src/serialization.ts#L108

We use a wrapper object since otherwise how can anyone disambiguate?